Clangen/v4.2.0 upstream

clangen opened the pull request:
wingui patches that will never be accepted upstream

clangen commented:
Was trying to create a PR against my own repo. Ignore.

Bill-Gray commented:
OK,  I know you said 'ignore'.  But the DPI-aware font scaling seems like a good idea,  which is causing me to look at other things you've got here.
- "Work around for #28 (Alt-Enter gets 'swallowed') : I see I eventually came up with a fix for that,  which (I think) addresses other problems as well.  I also ran into the 'compiler warnings' about `_UNICODE` and `UNICODE` not being #defined,  and fixed those.
- Being able to set the font is a nice idea.  It would be nice to have it in a somewhat cross-platform method;  i.e.,  ask for 10-point GIyxizib,  and you'll get that font with the platforms that support font choice (WinGui,  SDL1/2,  DOSVGA,  X11).  With obvious caveats : Glyxizib may not be available on that platform.
- You are probably not alone in wanting the menu bar display to be changeable.
- "Fixed bug where defaults size wasn't loaded properly" : I don't quite see what you're aiming at here.  The purpose of the way I had it was so that If `PDC_resize_screen()` were called before `initscr()`,  then the specified size would be used at startup.  Thus,  you could tell PDCurses that you wanted,  say,  a 90x34 screen right from the get-go.  It even works on platforms that don't normally support programmatic resizing (i.e.,  you can set a 90x34 screen before `initscr()`,  but can't reset the screen size once it's displayed.)
More later...

clangen commented:
Just for context: I've maintained my own fork of your repo for a couple years now and have hack/patched a few bits of functionality that I want in my app, and fixed a few bugs here or there (some of which may have been eventually fixed in other ways upstream, in your repo).

1. DPI-awareness: I had originally submitted this patch as a pull request against your repo quite a while ago, but it seems to have been closed out and superseded by this one that's still open: https://github.com/Bill-Gray/PDCurses/pull/83
2. Regarding my fix for #28: nice to know, I'll see about reverting my patch to see if it's no longer necessary.
3. Setting the default font: yeah, my solution isn't robust across platforms, I just wanted a quick way to automatically deuglify the app in older versions of Windows. :)
4. The size-related fix: PDCurses stores the last window dimensions in the registry, then reloads it automatically on startup. At some point reloading the default size from the registry wasn't working properly due to an order-of-operations issue, so I fixed it in my branch. You may have fixed this upstream at some point later in a different way, but this was required so the window size didn't get reset every launch.
